### PR TITLE
Assorted changes to support Raster Attribute Table in GeoServer

### PR DESCRIPTION
**Pull request by aaime**
Bulk of the work is GEOT-7478, but a few extra fixes are required to support the incoming Raster Attribute Table GeoServer community module. To build properly this requires a new release of imageio-ext release based on https://github.com/geosolutions-it/imageio-ext/pull/293 
(hence the SNAPSHOT dependency in this PR, will be turned into a release once all the pieces fit toghether)

# Checklist

- [x] I have read the [contribution guidelines](https://github.com/geotools/geotools/blob/main/CONTRIBUTING.md).
- [x] I have sent a [Contribution Licence Agreement](https://docs.geotools.org/latest/developer/procedures/contribution_license.html) (not required for small changes, e.g., fixing typos in documentation).
- [x] First PR targets the `main` branch (backports managed later; ignore for branch specific issues).
- [ ] Avoid [Java 9+ split packages](http://tutorials.jenkov.com/java/modules.html#split-packages-not-allowed).
- [ ] All the build checks are green ([see automated QA checks](https://docs.geotools.org/latest/developer/conventions/code/qa.html)).

For core and extension modules:

- [x] New unit tests have been added covering the changes.
- [ ] [Documentation](https://github.com/geotools/geotools/tree/main/docs) has been updated (if change is visible to end users).
- [x] There is an issue in [GeoTools Jira](https://osgeo-org.atlassian.net/projects/GEOT) (except for changes not visible to end users). 
- [ ] Commit message(s) must be in the form ``[GEOT-XYZW] Title of the Jira ticket``.
- [ ] Bug fixes and small new features are presented as a single commit.
- [ ] The commit targets a single objective (if multiple focuses cannot be avoided, each one is in its own commit, and has a separate ticket describing it).

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -476,11 +476,11 @@
     <db2.jdbc.version>11.5.6.0</db2.jdbc.version>
     <eclipse.emf.version>2.15.0</eclipse.emf.version>
     <elasticsearch.version>7.14.0</elasticsearch.version>
-    <imageio.ext.version>1.4.7</imageio.ext.version>
+    <imageio.ext.version>1.4-SNAPSHOT</imageio.ext.version>
```
Comment (dromagnoli) [inline pom.xml:479]: using SNAPSHOT versions here Is intentional ?

File: pom.xml [hunk 2]
```diff
@@ -476,11 +476,11 @@
...
     <informix.jdbc.version>4.50.7.1</informix.jdbc.version>
     <jackson2.databind.version>2.15.2</jackson2.databind.version>
     <jackson2.version>2.15.2</jackson2.version>
-    <jaiext.version>1.1.24</jaiext.version>
+    <jaiext.version>1.1-SNAPSHOT</jaiext.version>
```
Comment (dromagnoli) [inline pom.xml:483]: same as above

**Review by dromagnoli — APPROVED** [review]
looks good to me.
only a small questions on dependency versions

**Comment by aaime** [comment]
Hi Daniele, nope, the versions will be updated to use releases, I'm working on them

**Pull request merged**